schema/rdl_schema_0.1.json

```
@@ -1710,13 +1552,502 @@
+              "$ref": "#/$defs/Vulnerability_function"
+            },
+            "fragility": {
+              "title": "Fragility function",
+              "type": "object",
+              "description": "The fragility function used to calculate the impact of the hazard.",
+              "$ref": "#/$defs/Fragility_function"
+            },
+            "damage_to_loss": {
+              "title": "Damage-to-loss function",
+              "type": "object",
+              "description": "The damage-to-loss function (or damage-to-loss model) used to calculate the impact of the hazard in conjunction with a fragility function.",
+              "$ref": "#/$defs/Damage_to_loss_function"
+            },
+            "engineering_demand": {
+              "title": "Engineering demand function",
+              "type": "object",
+              "description": "The engineering damand function used to calculate the impact of the hazard.",
+              "$ref": "#/$defs/Engineering_demand_function"
+            }
+          },
+          "minProperties": 1
+        },
+        "analysis_details": {
+          "title": "Analysis details",
+          "type": "string",
+          "description": "Additional details about the analysis used to produce the vulnerability function used in the modelled scenario(s).",
+          "minLength": 1
+        },
+        "se_category": {
+          "title": "Socio-economic index category",
+          "type": "object",
+          "description": "The category used to classify the socio-economic category vulnerable to the hazard.",
+          "$ref": "#/$defs/Se_category"
+        }
+      },
+      "minProperties": 1
+    },
+    "Vulnerability_function": {
+      "title": "Vulnerability function",
+      "type": "object",
+      "description": "Details of a vulnerability function used to calculate the impact of the hazard.",
+      "properties": {
+        "approach": {
+          "title": "Impact function approach",
+          "type": "string",
+          "description": "The approach the impact function is based upon, taken from the closed [function approach codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#function_approach).",
+          "enum": [
+            "analytical",
+            "empirical",
+            "hybrid",
+            "judgement"
+          ],
+          "codelist": "function_approach.csv",
+          "openCodelist": false
+        },
+        "relationship": {
+          "title": "Impact relationship type",
+          "type": "string",
+          "description": "The type of function relationships used to calculate the impact values, taken from the closed [relationship type codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#relationship_type).",
+          "enum": [
+            "discrete values",
+            "mathematical function (bespoke)",
+            "mathematical function (parametric)"
```

Update per values in CSV file:

```suggestion
            "discrete",
            "math_bespoke",
            "math_parametric"
```